rasa_nlu/components.py

```diff
@@ -345,16 +356,18 @@ def __init__(self, use_cache: bool = True) -> None:
         self.component_cache = {}
 
     def __get_cached_component(self,
-                               component_name: Text,
+                               index: int,
                                model_metadata: 'Metadata'
                                ) -> Tuple[Optional[Component], Optional[Text]]:
         """Load a component from the cache, if it exists.
 
-        Returns the component, if found, and the cache key."""
-        from rasa_nlu import registry
+        Returns the component, if found, and the cache key.
+        """
 
+        from rasa_nlu import registry
+        component_name = model_metadata.component_name(index)
         component_class = registry.get_component_class(component_name)
-        cache_key = component_class.cache_key(model_metadata)
+        cache_key = component_class.cache_key(index, model_metadata)
```

Comment:
same here for the `cache_key` function, can we pass in only the  components metadata, and the whole metadata as a separate kwarg